python_modules/dagster/dagster/components/resolved/model.py

```diff
@@ -83,6 +83,19 @@ def from_model(fn: Callable[["ResolutionContext", Any], Any], **kwargs):
         """Resolve this field by invoking the function which will receive the entire parent model."""
         return Resolver(ParentFn(fn), **kwargs)
 
+    @staticmethod
+    def union(*resolvers: "Resolver"):
+        return Resolver(
+            lambda context, field_value: next(
+                (
+                    r.fn.callable(context, field_value)
+                    for r in resolvers
+                    if r.fn.callable(context, field_value)
+                ),
+                None,
+            ),
```

Comment:
hm so the strategy is to just try all the resolvers and pick the first one that return something? what if one throws?

Reply:
Hmm, yeah, good call - updated to more gracefully handle this

Ideally we'd disambiguate based on the model type, but I'm realizing there's not a super convenient way to get that from the resolver alone, do you have any thoughts? 

https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/components/resolved/model.py#L90

Reply:
https://github.com/dagster-io/dagster/pull/30014/files#diff-f080e3a990e91bb70a33f5c4ac215e90ea3149cac8052330c6d519dfaeab28b4R90